aea/configurations/validation.py

```diff
@@ -314,7 +315,20 @@ def check_excludes(path: Tuple[str, ...]) -> bool:
             continue
 
         if path not in pattern_path_value:
-            errors.append(f"Attribute `{'.'.join(path)}` is not allowed to be updated!")
+            is_a_dict_override = False
+            path_ = deepcopy(path)
+            while len(path_) > 0:
+                path_ = path_[:-1]
+                if path_ not in pattern_path_value:
+                    continue
+                pattern_value = pattern_path_value[path_]
+                is_a_dict_override = isinstance(pattern_value, OrderedDict)
+
+            if not is_a_dict_override:
```

Comment:
Why do we support the recursive approach only for dict overrides?

Reply:
try using proper recursion instead of iterative with deepcopy, flags and whatnot. Keeps the recursion nicely separate from the other logic.
```
def rec_check_pattern_in_path(path) -> bool:
    return path in pattern_to_path or rec_check_pattern_in_path(path[1:])
```